sympy/solvers/ode.py

```diff
@@ -704,7 +704,11 @@ def _helper_simplify(eq, hint, match, simplify=True, ics=None, **kwargs):
     else:
         # We still want to integrate (you can disable it separately with the hint)
         match['simplify'] = False  # Some hints can take advantage of this option
-        rv = _handle_Integral(solvefunc(eq, func, order, match), func, hint)
+        exprs = solvefunc(eq, func, order, match)
+        if isinstance(exprs, list):
+            rv = [_handle_Integral(expr, func, hint) for expr in exprs]
+        else:
+            rv = _handle_Integral(exprs, func, hint)
```

Comment:
We really need to clean it up so that solvefunc always returns a list. We shouldn't get into this situation where we don't know if we have a list or not.

Reply:
I need to open a new pr for this because initially dsolve is made in a way that mostly it will return a single equation solution. So almost all of the method return an equation. So I have to change this for many of the methods.  For this pr We can go in this way and later I will change it.